src/main/java/backend/Logic.java

```diff
@@ -331,6 +331,30 @@ private void revertLocalLabelsReplace(TurboIssue modifiedIssue, List<String> ori
     }
 
     /**
+     * Edit the state of an issue on GitHub
+     *
+     * @param issue The issue whose state is to be updated
+     * @param open The new state for the issue
+     * @return True for success, false otherwise
+     */
+    public CompletableFuture<Boolean> editIssueState(TurboIssue issue, boolean open) {
+        String action = open ? "Reopening" : "Closing";
+        logger.info(String.format("%s %s on GitHub", action, issue));
+        return repoIO.editIssueState(issue, open)
+                .thenApply(success -> {
+                    if (success) {
+                        logger.info("Changing state for " + issue + " on UI");
+                        issue.setOpen(open);
+                        refreshUI();
+                        return true;
+                    } else {
+                        return false;
```

Comment:
The `else` clause here is not necessary. The preferred controlled flow is to deal with the exceptional cases first e.g.
```java
if (!isSuccessful) {
  return false;
}
// Do the main successful flow
....
```